library/core/src/iter/adapters/dedup.rs

```diff
@@ -50,7 +50,9 @@ where
     }
 
     fn size_hint(&self) -> (usize, Option<usize>) {
-        (0, self.inner.size_hint().1)
+        let min = self.last.as_ref().map(|_| 1).unwrap_or(0);
+        let max = self.inner.size_hint().1;
```

Comment:
You should add lower bound to size_hint, probably.
You now can end with situation `(1, Some(0))`.

Reply:
Oh yes, you're right